File: src/main/kotlin/racingcar/domain/Car.kt
package racingcar.domain

import kotlin.random.Random

class Car(val name : String){
    var position : Int = 0
        private set
    init {
        require(name.length <= CAR_NAME_LENGTH) { "자동차 이름은 5글자를 초과할 수 없습니다." }
        require(!name.contains(" ")) { "자동차 이름은 공백을 포함 할 수 없습니다." }
    }
    fun move(){
        position++
    }
    companion object {
        private const val CAR_NAME_LENGTH = 5
    }
}

File: racingcar/racingcar/domain/Car.swift
//
//  Car.swift
//  racingcar
//
//  Created by KimMinSeok on 3/26/24.
//
enum CarInputError: Error {
    case invalidArgumentException
}

import Foundation

class Car {
    private(set) var name: String
    var position = 0
    let moveStrategy : MoveStrategy
    
    init(name: String, position: Int = 0, moveStrategy: MoveStrategy) throws {
        self.name = name
        self.position = position
        self.moveStrategy = moveStrategy
        
        // 이름이 기준에서 넘거나 빈 문자열로 들어온 경우 예외처리
        if !(Car.minNameLength...Car.maxNameLength).contains(name.count) || name.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            throw CarInputError.invalidArgumentException
        }
    }

    func go() {
        if(moveStrategy.canMove()){
            position += 1
        }
    }
    
    static let minNameLength = 1
    static let maxNameLength = 5
}
